module/modmanager/manager.go

```diff
@@ -758,15 +758,35 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	ctxTimeout, cancelFunc := context.WithTimeout(ctx, rutils.GetModuleStartupTimeout(logger))
 	defer cancelFunc()
 
+	slowTicker := time.NewTicker(15 * time.Second)
+	defer slowTicker.Stop()
+	startTime := time.Now()
+
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-slowTicker.C:
+				elapsed := time.Since(startTime).Seconds()
+				logger.Warnf("waiting %q for module to be ready. Elapsed %.2f seconds", m.cfg.Name, elapsed)
+			case <-done:
```

Comment:
Instead of a `done` channel can you just `case <-ctxTimeout.Done():` here? That should exit the goroutine when the function ends (because of the deferred `cancelFunc` above).